### PR TITLE
Use take_1d in take_nd

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
xref https://github.com/pandas-dev/pandas/pull/40300#issuecomment-797637165

### CODE REVIEW

**Comment by jbrockmendel** [comment]
can remove comment in take_1d about being only for ArrayManager

**Comment by jbrockmendel** [comment]
test failure is unrelated (and the flaky test is fixed on master).  LGTM

**Review by jbrockmendel — APPROVED** [review]

**Comment by jreback** [comment]
cool, does this allow refactor of other code?

**Comment by jbrockmendel** [comment]
can this be moved out of Draft mode?

**Comment by jbrockmendel** [comment]
@jorisvandenbossche gentle ping

**Comment by jbrockmendel** [comment]
@jorisvandenbossche gentle ping to see if we can move this out of draft mode

**Comment by jbrockmendel** [comment]
@jorisvandenbossche gentle ping

**Comment by jorisvandenbossche** [comment]
Updated the PR. 
But personally not sure if it is worth it, though:

```python
In [1]: from pandas.core.array_algos.take import take_1d, take_nd

In [2]: dtype = 'int'
   ...: m = 100
   ...: n = 1000

In [3]: values = np.arange(m * n)

In [4]: index = np.arange(m)

# this is not affected, just including as reference
In [5]: %timeit take_1d(values, index)
5.67 µs ± 27 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)  <-- master
5.59 µs ± 50.6 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)  <-- PR

In [6]: %timeit take_nd(values, index)
10.5 µs ± 104 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)  <-- master
8.86 µs ± 68.3 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)  <-- PR
```

So it does give a small improvement, but when increasing the size of the array, the improvement almost completely disappears.

**Comment by jbrockmendel** [comment]
3 lines for a  15% improvement for a fairly common case seems like a good deal to me

File: pandas/core/array_algos/take.py [hunk 1]
```diff
@@ -89,6 +89,10 @@ def take_nd(
     if fill_value is lib.no_default:
         fill_value = na_value_for_dtype(arr.dtype, compat=False)
 
+    if arr.ndim == 1 and axis == 0 and indexer is not None:
+        indexer = ensure_platform_int(indexer)
```
Comment (jbrockmendel) [inline pandas/core/array_algos/take.py:93]: comment to the effect of "fastpath"

**Comment by simonjayhawkins** [comment]
Thanks @jorisvandenbossche for the PR. closing as stale. re-open when ready.

**Pull request closed without merging**